### PR TITLE
Various minor fixes: phpdoc, type-hints, imports

### PR DESCRIPTION
**Pull request by mnapoli**
Hi there, first contribution here :) No a big change, but I opened the project in PhpStorm and there were a few warnings here and there:
- wrong on missing phpdoc
- missing type-hints that would help autocompletion and static analysis
- unused class imports
- unused variables

I have fixed some of them, I didn't add extensive phpdoc (which would help even more in autocompletion and static analysis) because I wasn't sure if it was OK regarding the style guide of the project.

### CODE REVIEW

File: Slim/Autoloader.php [hunk 1]
```diff
@@ -19,7 +19,6 @@ public static function autoload($className)
     {
         $className = ltrim($className, '\\');
         $fileName  = '';
-        $namespace = '';
```
Comment (mnapoli) [inline Slim/Autoloader.php:22]: unused outside of the `if`, immediately overwritten inside the `if`

File: Slim/Http/Request.php [hunk 2]
```diff
@@ -308,7 +307,7 @@ public function withMethod($method)
      *
      * @param  null|string $method
      * @return null|string
-     * @throws InvalidArgumentException on invalid HTTP method.
```
Comment (mnapoli) [inline Slim/Http/Request.php:311]: without the `\` it was resolved in the current namespace

File: Slim/Http/Uri.php [hunk 3]
```diff
@@ -284,7 +284,6 @@ protected function filterScheme($scheme)
      */
     public function getAuthority()
     {
-        $scheme = $this->getScheme();
```
Comment (mnapoli) [inline Slim/Http/Uri.php:287]: unused

File: Slim/Interfaces/Http/HeadersInterface.php [hunk 4]
```diff
@@ -8,8 +8,6 @@
  */
 namespace Slim\Interfaces\Http;
 
-use \Slim\Interfaces\Http\CollectionInterface;
```
Comment (mnapoli) [inline Slim/Interfaces/Http/HeadersInterface.php:11]: not necessary, same namespace

File: Slim/Router.php [hunk 5]
```diff
@@ -23,13 +23,15 @@ class Router extends \FastRoute\RouteCollector implements RouterInterface
...
+     *
+     * @var Route[]
      */
     protected $routes = [];
 
     /**
      * Named routes
      *
-     * @var null|array
+     * @var null|Route[]
```
Comment (mnapoli) [inline Slim/Router.php:34]: I wasn't sure if it was OK with the project's coding style (I can remove this change), but documenting like that offers autocompletion and static analysis when iterating the array, which is pretty useful to detect potential errors or to navigate the code (Ctrl+click on a method call for example).

**Comment by codeguy** [comment]
Thank you!

**Pull request merged**